web/pgadmin/static/js/tree/tree_nodes.ts

```diff
@@ -60,6 +60,13 @@ export class ManageTreeNodes {
   public addNode = (_parent: string, _path: string, _data: []) => new Promise((res) => {
     _data.type = _data.inode ? FileType.Directory : FileType.File;
     _data._label = _data.label;
+
+    const nodeClass = pgAdmin.Browser.Nodes[_data._type];
+
+    if (nodeClass && typeof nodeClass.getNodeLabel === 'function') {
```

Comment:
I don't think we need to check `getNodeLabel`. Plus you've already provided the default value.